extensions/modules/src/org/exist/xquery/modules/xslfo/ApacheFopProcessorAdapter.java

```diff
@@ -126,7 +126,9 @@ private FOUserAgent setupFOUserAgent(final FOUserAgent foUserAgent, final Proper
      */
     private ResourceResolver getResourceResolver(final DBBroker broker, final String baseUri) {
         final ResourceResolverFactory.SchemeAwareResourceResolverBuilder builder = ResourceResolverFactory.createSchemeAwareResourceResolverBuilder(ResourceResolverFactory.createDefaultResourceResolver());
-        final URIResolverAdapter uriResolver = new URIResolverAdapter(new ExistSchemeRewriter(new EXistURIResolver(broker, baseUri)));
+        final URIResolverAdapter uriResolver = new URIResolverAdapter(
+            new ExistSchemeRewriter(new EXistURIResolver(broker.getBrokerPool(), baseUri))
+        );
         builder.registerResourceResolverForScheme("exist", uriResolver);
```

Comment:
Okay but I think this is a bigger issue that we should all agree. We should at one of our teleconferences have a read and build a proposal for a unified URL scheme that is compatible with java.net.URI and also how we can support backwards compatibility with `xmldb:exist:/` for a period of time.